client/src/context/Reservation/index.js

```diff
@@ -0,0 +1,71 @@
+import { createContext, useReducer, useState } from 'react';
+import { getLocalStorage } from './localStorage';
+import cinemaHallMock from '../../mock/cinemaHallMock';
+
+export const ReservationContext = createContext();
+const initialState = {
+  screeningId: '',
+  cinemaHall: cinemaHallMock[0],
+  movieId: '',
+  totalTickets: 0,
+  selectedSeats: [],
+};
+const localReservation = getLocalStorage('reservation', initialState);
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'SUCCESS_CINEMAHALL':
+      return { cinemaHall: action.payload };
+    case 'ADD_SCREENING_ID':
+      return {
+        screeningId: action.payload,
+      };
+    case 'ADD_CINEMAHALL_ID':
+      return {
+        cinemaHallId: action.payload,
+      };
+    case 'ADD_MOVIE_ID':
+      return {
+        movieId: action.payload,
+      };
```

Comment:
Are these necessary?

Reply:
It was Your idea :). I don't need movie id to CinemaHall

Reply:
I mean you have them in the context after getting all the info about screening. Do you need them in the reducer? maybe just payload would be enough than separating into individual actions.